packages/react-native/ReactCommon/react/nativemodule/core/platform/ios/ReactCommon/RCTTurboModule.mm

```diff
@@ -400,12 +471,47 @@ id convertJSIValueToObjCObject(jsi::Runtime &runtime, const jsi::Value &value, s
       TurboModulePerfLogger::asyncMethodCallExecutionStart(moduleName, methodNameStr.c_str(), asyncCallCounter);
     }
 
-    @try {
+    try {
       [inv invokeWithTarget:strongModule];
-    } @catch (NSException *exception) {
-      throw convertNSExceptionToJSError(runtime, exception);
-    } @finally {
-      [retainedObjectsForInvocation removeAllObjects];
```

Comment:
Keep the retainedObjectsForInvocation in a finally block, instead of handling it inside `handleCause`

Reply:
I switch from `@try` to `try` and that doesn't have `finally` in C++. That's why I added in to the `handleCause`.